### DISCUSSION TITLE
On the use of Incompact3D for periodic hill

### DISCUSSION THREAD

**Discussion by git-ywang** [xcompact3d/Incompact3d#186] [Q&A] [answered]
Dear code developers and researchers,
Recently I am trying to run a DNS for periodic hill (PH), and having the following questions on the setup of the initial and boundary conditions: 

1. According to the paper: "_Ahlfeld, R., Laizet, S.,  Montomoli, F., Geraci, G., and  Iaccarino, G. Multi-Fidelity Uncertainty Quanti cation Using RANS and DNS_", one should use a parabolic profile initially until fully developed channel flow is obtained, and subsequently, incorporate the periodic hills to the domain.  So, is this a must-follow percedure to simulate PH flow with Incompact3d? Or is it possible to simulate the flow in one shot from the beginning?
 
2. Following the first question, If I also have to run the channel flow first and then switch to PH flow,  do I need to switch the mesh when the hills are incorporated? since the meshes are usually finer near the upper and lower walls in channel flow, but the mesh should then become finer near the hills' surfaces once they are added to the domain. Are there any other changes that need to be made in addition to switch the flow type to PH and turn on IBM?

3. The last question is about the parallel implementation of the code. When I test the code for PH flow, segmentation fault occasionally occurs depending on how I configure the parallel computational cores, which is really intriguing. For example, when I use 64 cores, an 8*8 (p_row*p_col) configuration would cause the segmentation fault, while a 4*16 or 16*4 configuration runs normally.  Is this sth normal for the code?

I appreciate helps of any kind !

**Comment by slaizet** [comment, marked as answer]
Hi,
We are going to release in the next few days the files to run the periodic 2d hill case with an input file.
Best,
Sylvain
PS: very strange behaviour regarding your question 3; this should not happen...

**Reply by git-ywang** [reply]
Okay ! Thank you !

**Reply by slaizet** [reply]
You should now see an input file for the periodic 2d hill in a new example folder.
Best,
Sylvain

**Reply by git-ywang** [reply]
Sure. Thank you for letting me know ! That will be of great help !
